File: app/src/main/java/com/softhink/single/GenericObserver.java
package com.softhink.single;

import com.softhink.single.registro.Status;

import androidx.lifecycle.Observer;

public class GenericObserver<T> implements Observer<T> {
    private Status status;
    private T t;

    public GenericObserver(Status status, T t) {
        this.status = status;
        this.t = t;
    }

    @Override
    public void onChanged(T t) {
    }

    public Status getStatus(){
        return status;
    }

    public T getData(){
        return t;
    }
}

File: app/src/main/java/com/softhink/single/data/manager/GenericObserver.kt
package com.softhink.single.data.manager

import com.softhink.single.ui.registro.Status
import androidx.lifecycle.Observer

class GenericObserver<T>(var status: Status, val data: T?, var message: String?) : Observer<T> {

    override fun onChanged(t: T) {}
}
